### ISSUE TITLE
bug(@angular/compiler-cli): Unable to handle Parallel execution

### ISSUE THREAD

**Issue by FrozenPandaz** [angular/angular#14927]
**I'm submitting a ...**
```
[x] bug report
```

**Current behavior**
The compiler-cli has interesting side effects when run in parallel.

This is commonly done when using it through @ngtools in multiple parallel webpack configs

**Expected behavior**
The compiler-cli should be capable of being executed in parallel by the same node process.

At the very lease there should be an error or warning in place to explain that it does not have that capability yet

**Minimal reproduction of the problem with instructions**
This tree shows the side effects: https://github.com/FrozenPandaz/ng-universal-demo/commit/ebf40c7b56c69784f0ed59955972efad3b0dde54

You can play around the webpack config export (last line).

Any of the configs provided by themselves will work.

Any combination of the configs provided will have weird side effects for one of the builds.

**What is the motivation / use case for changing the behavior?**
The motivation is that the compiler-cli should be robust enough to run in parallel.

The common use case is for universal apps which want to run two webpack builds in parallel. One to generate the client bundle and another to generate the server bundle

Once this is possible we can even compile in parallel and improve performance?

**Please tell us about your environment:**
NodeJS

* **Angular version:** 2.0.X
2.0+
It's still an issue on 4.0.0-rc.2

* **Browser:** [all | Chrome XX | Firefox XX | IE XX | Safari XX | Mobile Chrome XX | Android X.X Web Browser | iOS XX Safari | iOS XX UIWebView | iOS XX WKWebView ]
Not applicable
 
* **Language:** [all | TypeScript X.X | ES6/7 | ES5]
Typescript

* **Node (for AoT issues):** `node --version` =   
6.10.x

**Comment by FrozenPandaz** [comment]
@chuckjaz, @hansl said he spoke to you about this.

**Comment by FrozenPandaz** [comment]
The workaround is [running the webpack configs in sequence](https://github.com/FrozenPandaz/ng-universal-demo/blob/master/package.json#L9).

Are we able to hotfix this temporarily to just have the @ngtools/webpack plugin to wait for an async signal that there are no other compilers in flight?

**Comment by hansl** [comment]
@FrozenPandaz, or run two webpack processes. Should be relatively simple. I'm not sure re-using the Compiler CLI twice for two different projects in the same process is going to work, but I don't know for sure. Also, watch mode will stay slow if we run sequentially, but relatively fast in 2 separate process.

**Comment by FrozenPandaz** [comment]
@hansl Good point, they dont have to be sequential but they do have to be in different node processes. Running them separated by ```&``` works.

```
"build:aot": "webpack --env.aot --env.client & webpack --env.aot --env.server"
```

**Comment by alxhub** [comment]
Multiple instances of the View Engine compiler in the same process wasn't a design goal, I believe.

For Ivy, this was not explicitly considered either, but due to the architecture it should work, in theory.

**Comment by NoUsername** [comment]
this is still an issue, #32431 also has more details about this.

especially in larget projects which may get split up into separate projects (but within a monorepo with a single build) this is almost becoming a blocker, with each new angular project that is added, the build time shoots up again.

**Comment by jessicajaniuk** [comment]
We expect this to be resolved at this point, but if it's not, please open a new ticket. See [this comment](https://github.com/angular/angular/issues/32431#issuecomment-603818057) for more information.
